### DISCUSSION TITLE
ios and Xcode build support

### DISCUSSION THREAD

**Discussion by debpalash** [robertodoering/harpy#702] [General]
I've cloned and enabled this project for ios, hardcoded `environment.dart` with consumer key and secrets but the launchScreen throws " Invalid Twitter api/ secret".

```
final environmentProvider = Provider(
  (ref) {
    const aesKey = String.fromEnvironment('aes_key');
    const twitterConsumerKey = String.fromEnvironment(
      'qtdoP831o3kPtd5YgRxxxxxx',
    );
    const twitterConsumerSecret = String.fromEnvironment(
      '4qj2CfDNXDma0vJqrEwDlTA4Q3TxxxxxxxQ7eGrxPUfz',
    );
```

after running -flavor commands there's a erorr of .xcscheme support with -flavor for xcode.

Has anyone's build for ios succeed? what am i doing wrong?

**Comment by robertodoering** [comment]
For the keys, don't use `String.fromEnvironment` if you want to hardcode them, just set them directly
```dart
const twitterConsumerKey = 'qtdoP831o3kPtd5YgRxxxxxx';
const twitterConsumerSecret =  '4qj2CfDNXDma0vJqrEwDlTA4Q3TxxxxxxxQ7eGrxPUfz';
```

Instead of building with a `flavor`, you can just omit the flag and hardcode the current flavor in `environment.dart`.

**Reply by killabit** [reply]
Can you elaborate a little on this?  I literally have been fighting XCode for like 9 hours straight it keeps failing on transparent images, so I just took it to XCode Cloud, and it literally builds all the way but failed on the last step which was a push to origin of "Cleanup" 

Apple is so frustrating..

**Comment by killabit** [comment]
Holy crap I finally got it to build, building for iOS is not streamlined at all what a crazy adventure..
